File: app/src/main/java/com/smartwavettn/horoscope/broadcast/AlarmReceiver.java
package com.smartwavettn.horoscope.broadcast;

import android.Manifest;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;

import androidx.core.app.ActivityCompat;
import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;

import com.smartwavettn.horoscope.R;
import com.smartwavettn.horoscope.ui.MainActivity;

public class AlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {

        Intent i = new Intent(context, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

        PendingIntent pendingIntent = PendingIntent.getActivity(context,0,i, PendingIntent.FLAG_IMMUTABLE);

        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, "channelId")
                .setSmallIcon(com.smartwavettn.horoscope.R.drawable.aaaa)
                .setContentTitle(context.getString(R.string.alarm))
                .setContentText("conten ................................................")
                .setAutoCancel(true)
                .setDefaults(NotificationCompat.DEFAULT_ALL)
                .setPriority(NotificationCompat.PRIORITY_HIGH)
                .setContentIntent(pendingIntent);


        NotificationManagerCompat notificationManagerCompat = NotificationManagerCompat.from(context);
        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
            return;
        }
        notificationManagerCompat.notify(123, builder.build());
    }
}

File: app/src/main/java/com/smartwavettn/horoscope/broadcast/AlarmReceiver.kt
package com.smartwavettn.horoscope.broadcast

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.widget.RemoteViews
import androidx.core.app.NotificationCompat
import com.smartwavettn.horoscope.R

class AlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == "MyAction") {
            val notificationManager =
                context.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager
            if (notificationManager != null) {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    val channel = NotificationChannel("201", "Notification", NotificationManager.IMPORTANCE_HIGH)
                    channel.description = "Lovely"
                    notificationManager.createNotificationChannel(channel)
                }

                val randomMessage = "getRandomMessage(context)"

                val contentView = RemoteViews(context.packageName, R.layout.notification_layout)
                contentView.setTextViewText(R.id.notification_title, "Daily")
                contentView.setTextViewText(R.id.notification_message, randomMessage)


                val builder = NotificationCompat.Builder(context, "201")
                    .setSmallIcon(R.drawable.notification)
                    .setContent(contentView)
                    .setColor(Color.parseColor("#FF0000"))
                    .setCategory(NotificationCompat.CATEGORY_ALARM)

                notificationManager.notify(getNotificationId(), builder.build())
            }
        }
    }

    private fun getNotificationId(): Int {
        return System.currentTimeMillis().toInt()
    }

}